### PR TITLE
Clipping path fix for browsers that don't use correct units when applying clip-path css using svg

### PR DESCRIPTION
**Pull request by ferhatb**
Safari requires objectBoundingBox units on svg clipPaths to function correctly, the default units don't scale correctly on these browsers when page is zoomed or used on highdpi devices.
Fixes https://github.com/flutter/flutter/issues/48683

### CODE REVIEW

File: lib/web_ui/lib/src/engine/util.dart [hunk 1]
```diff
@@ -312,18 +312,21 @@ int _clipIdCounter = 0;
 /// Calling this method updates [_clipIdCounter]. The HTML id of the generated
 /// clip is set to "svgClip${_clipIdCounter}", e.g. "svgClip123".
 String _pathToSvgClipPath(ui.Path path,
-    {double offsetX = 0, double offsetY = 0}) {
+    {double offsetX = 0,
+    double offsetY = 0,
+    double scaleX = 1.0,
+    double scaleY = 1.0}) {
   _clipIdCounter += 1;
   final ui.Rect bounds = path.getBounds();
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/util.dart]: Unused variable?

File: lib/web_ui/test/golden_tests/engine/canvas_clip_path_test.dart [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    path.addOval(Rect.fromLTWH(100, 30, testWidth, testHeight));
+    rc.clipPath(path);
+    rc.drawImageRect(testImage, Rect.fromLTRB(0, 0, testWidth, testHeight),
+        Rect.fromLTWH(100, 30, testWidth, testHeight), Paint());
+    rc.restore();
+    await _checkScreenshot(rc, 'image_clipped_by_oval');
+  });
+
+  // Regression test for https://github.com/flutter/flutter/issues/48683
+  test('Clips triangle with oval clip path devicePixelRatio2', () async {
```
Comment (yjbanov) [inline lib/web_ui/test/golden_tests/engine/canvas_clip_path_test.dart]: What does "devicePixelRatio2" in the description mean? I wasn't able to parse it from the test itself.

File: lib/web_ui/lib/src/engine/util.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/util.dart]: Removed.

File: lib/web_ui/test/golden_tests/engine/canvas_clip_path_test.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/test/golden_tests/engine/canvas_clip_path_test.dart]: removed. Initially explored EngineWindow.debugDevicePixelRatio but snapshots still need some infra updates as discussed.

**Comment by ferhatb** [comment]
Merging on red, infra failures.

**Pull request merged**